File: 219. Contains Duplicate II/solution.java
package com.company;

import java.util.*;
import java.util.stream.Stream;

public class Main {


    public static void main(String[] args) {
        int nums[] = {1, 2, 3, 1, 2, 3};
        boolean b = containsNearbyDuplicate(nums, 2);
        System.out.println(b);
    }

    public static boolean containsNearbyDuplicate(int[] nums, int k) {
        for (int i = 0; i < nums.length; i++) {
            for (int j = i + 1; j < nums.length; j++) {
                if ((j - i) <= k) {
                    if (nums[j] == nums[i]) return true;
                } else {
                    break;
                }
            }
        }
        return false;
    }

}

File: 219. Contains Duplicate II/solution.cs
using System.Collections.Generic;
using System.Linq;
using System.Collections;
using System;
using System.Text;
using System.Net.Http;

namespace a
{
    public class Solution
    {
		
        public static void Main()
        {	
            int[] n = { 99, 99 };        
            var b = ContainsNearbyDuplicate(n, 3);
            Console.WriteLine(b);
            Console.ReadLine();
        }
        public static bool ContainsNearbyDuplicate(int[] nums, int k)
        {
            var map = new Dictionary<int,List<int>>();
            for (int i = 0; i < nums.Length; i++) {
                List<int> list = new List<int>();
                
                if (!map.ContainsKey(nums[i])) {
                    list.Add(i);
                    map.Add(nums[i], list);
                } else {
                    var listT = new List<int>();
                    map.TryGetValue(nums[i], out list);
                    list.Add(i);
                }
            }
            foreach (var kp in map) {
                var vList = kp.Value;
                for (int i = 0; i < vList.Count; i++) {
                    for (int j = i + 1; j < vList.Count; j++) {
                        var diffValue = Math.Abs((vList[i] - vList[j]));
                        if (diffValue <= k) {
                            return true;
                        } else if (diffValue > k) {
                            break;
                        }
                    }
                }
            }
            return false;
        }
		
		
				

    }
	   
}


